### PR TITLE
SCP-3501: Remove txs from chain index

### PR DESCRIPTION
**Pull request by ak3n**
Pre-submit checklist:
- Branch
    - [x] Tests are provided (if possible)
    - [x] Commit sequence broadly makes sense
    - [x] Key commits have useful messages
    - [ ] Relevant tickets are mentioned in commit messages
    - [ ] Formatting, materialized Nix files, PNG optimization, etc. are updated
- PR
    - [x] Self-reviewed the diff
    - [ ] Useful pull request description
    - [x] Reviewer requested

### CODE REVIEW

File: plutus-chain-index-core/src/Plutus/ChainIndex/Handlers.hs [hunk 1]
```diff
@@ -82,7 +82,6 @@ handleQuery = \case
     MintingPolicyFromHash hash  -> getScriptFromHash hash
     RedeemerFromHash hash       -> getRedeemerFromHash hash
     StakeValidatorFromHash hash -> getScriptFromHash hash
-    TxFromTxId txId             -> getTxFromTxId txId
     TxOutFromRef tor            -> getTxOutFromRef tor
```
Comment (koslambrou) [inline plutus-chain-index-core/src/Plutus/ChainIndex/Handlers.hs:85]: I would change this to `UnspentTxOutFromRef` and `getUtxoutFromRef`

**Review by koslambrou — CHANGES_REQUESTED** [review]
The implementation looks good. However, I would also add a property test to ensure this works as expected. Here's an idea.

In HandlerSpec, you can add a property test which:
- calls `appendBlock`
- for all possibles addresses (or at least one address), call `utxoSetAtAddress` 
- Then call `getUtxoutFromRef` for each utxo
- Verify the result is not Nothing

File: plutus-chain-index-core/src/Plutus/ChainIndex/Api.hs [hunk 2]
```diff
@@ -147,11 +146,9 @@ type API
     = "healthcheck" :> Description "Is the server alive?" :> Get '[JSON] NoContent
     :<|> "from-hash" :> FromHashAPI
     :<|> "tx-out" :> Description "Get a transaction output from its reference." :> ReqBody '[JSON] TxOutRef :> Post '[JSON] ChainIndexTxOut
```
Comment (koslambrou) [inline plutus-chain-index-core/src/Plutus/ChainIndex/Api.hs:148]: Change this as well to refer to unspent txouts only.

File: plutus-chain-index-core/src/Plutus/ChainIndex/Handlers.hs (hunk 1 above)
Comment (ak3n) [reply plutus-chain-index-core/src/Plutus/ChainIndex/Handlers.hs]: Should we rename everything in plutus-contract then?

Comment (koslambrou) [reply plutus-chain-index-core/src/Plutus/ChainIndex/Handlers.hs]: Yep!

**Review by koslambrou — APPROVED** [review]

**Pull request merged**